src/Microsoft.FeatureManagement/IsExternalInit.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+// The init accessor for properties is supported in C# 9.0 and later.
+// This class is used to compile .NET frameworks that don't support C# 9.0 or later while still using the init accessor for a property.
+// The code referenced for this file can be found here: https://github.com/dotnet/roslyn/issues/45510#issuecomment-725091019
+
+#if NETSTANDARD2_0 || NETCOREAPP2_1 || NETCOREAPP3_1
```

Comment:
variants haven't been merged. Perhaps this is no longer needed.

Reply:
I'm using this for the init of TelemetryPublisher https://github.com/microsoft/FeatureManagement-Dotnet/pull/259/files#diff-7726d09bdcf1556c447fe12c6bd0f83385916697784034cec5d959edf8e7021aR59 as it's also an optional dependency. 

This PR will be merged into `telemetry-preview` branch, and then a separate PR will merge into `preview` (deduping this class).

Reply:
What's your plan with these branches? I was expecting this merge directly to preview and us have a preview release with it.

Reply:
2 things:

1. I wanted a clear separation of what's being reviewed. This is a review of the telemetry pipeline (Goal 1 of the design doc). The next PR will be a merge from telemetry-preview to preview, which will include the variant logic as well. 
2. Variants weren't merged yet when I made this PR. 

If this was merged into preview now, there are code paths that wouldn't emit telemetry and conflicts.

However this does mean these changes will be reviewed again when including the variants and merging into preview.